File: backend/app/models/requirement.py
from pydantic import BaseModel
from typing import List, Optional


class Requirement(BaseModel):
    id: str
    name: str
    category: str
    description: str
    authority: str
    priority: str
    status: str = "pending"

    why_applies: List[str] = []

    documents: List[dict] = []

    application_process: Optional[str] = None

File: frontend/src/lib/requirement.ts
export interface Requirement {
  id: string;
  name: string;
  description?: string;
  category?: string;
  authority?: string;
  status?: string;
  applicability_reason?: string;
  reason?: string;
}

export interface RequirementDiscoveryResponse {
  business_id?: string;
  requirements?: Requirement[];
  matched_requirements?: Requirement[];
}